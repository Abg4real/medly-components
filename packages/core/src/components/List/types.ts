import { HTMLProps, WithThemeProp } from '@medly-components/utils';

export interface Props extends HTMLProps<HTMLUListElement>, WithThemeProp {
    variant?: 'horizontal' | 'vertical';
}
