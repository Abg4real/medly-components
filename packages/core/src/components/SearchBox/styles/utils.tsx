import { SvgIcon } from '@medly-components/icons';
import { css } from '@medly-components/utils';
import { Props } from '../types';

export const getSearchBoxSize = ({ theme, size }: Props) => theme.searchBox.sizes[size].height;

export const getBorderAndBoxShadow = (color: string, boxShadow: string) => {
    return css`
        border-color: ${color};
        box-shadow: ${boxShadow};
    `;
};

export const getIconWrapperStyle = ({ theme, size }: Props) => {
    return css`
        width: ${getSearchBoxSize};
        min-width: ${getSearchBoxSize};
        height: auto;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        ${SvgIcon} {
            cursor: pointer;
            font-size: ${theme.searchBox.sizes[size].iconSize};
        }
    `;
};
