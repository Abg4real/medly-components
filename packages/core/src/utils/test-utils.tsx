import { defaultTheme } from '@medly-components/theme';
import React from 'react';
import { render, RenderResult } from 'react-testing-library';
import { ThemeProvider } from './styled';

const AllTheProviders: React.FunctionComponent = props => (
    <ThemeProvider theme={defaultTheme}>
        <>{props.children}</>
    </ThemeProvider>
);

const customRender = (ui: React.ReactElement<any>, options?: object): RenderResult => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from 'react-testing-library';

// override render method
export { customRender as render };
