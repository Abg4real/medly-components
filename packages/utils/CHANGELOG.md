# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [1.0.0-beta.16](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.15...@medly-components/utils@1.0.0-beta.16) (2020-04-11)


### Features

* add typography and refactor components based on new typography ([462f2aa](https://github.com/medlypharmacy/medly-components/commit/462f2aa5b24c968b941887e6fcea5fd4e5c7b3f1))





# [1.0.0-beta.15](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.14...@medly-components/utils@1.0.0-beta.15) (2020-04-02)


### Bug Fixes

* **all:** fix lint issues ([ca12080](https://github.com/medlypharmacy/medly-components/commit/ca1208092c7a4d0ce0e22f20cb31c0400c4ddbff))





# [1.0.0-beta.14](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.13...@medly-components/utils@1.0.0-beta.14) (2020-03-30)


### Features

* **all:** add rollup to create bundles ([08b5972](https://github.com/medlypharmacy/medly-components/commit/08b5972f4a221037d14449ffce508f8656d0f447))





# [1.0.0-beta.13](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.12...@medly-components/utils@1.0.0-beta.13) (2020-03-29)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.12](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.11...@medly-components/utils@1.0.0-beta.12) (2020-02-06)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.11](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.10...@medly-components/utils@1.0.0-beta.11) (2020-02-04)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.10](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.9...@medly-components/utils@1.0.0-beta.10) (2020-01-22)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.9](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.8...@medly-components/utils@1.0.0-beta.9) (2020-01-21)


### Features

* **core:** add functionaity to accept date as string also in datepicker component ([3c3c377](https://github.com/medlypharmacy/medly-components/commit/3c3c377a66c470e6765d4a98f0f55b443b5ab0bc))
* **icons:** add bill icon ([2b461c2](https://github.com/medlypharmacy/medly-components/commit/2b461c2478590247498b08f13cad83d225efff6e))





# [1.0.0-beta.8](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.7...@medly-components/utils@1.0.0-beta.8) (2020-01-12)


### Features

* **all:** add side-effects to package.json ([0f78b4c](https://github.com/medlypharmacy/medly-components/commit/0f78b4c2ba432373b9d66861c25a265dacd5d433))





# [1.0.0-beta.7](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.6...@medly-components/utils@1.0.0-beta.7) (2020-01-09)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.6](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.5...@medly-components/utils@1.0.0-beta.6) (2019-12-17)


### Bug Fixes

* **core:** fix modal close issue ([f2d8fe9](https://github.com/medlypharmacy/medly-components/commit/f2d8fe9553f9efdbeb96c8b5b213844f0c472777))





# [1.0.0-beta.5](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.4...@medly-components/utils@1.0.0-beta.5) (2019-12-10)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.4](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.3...@medly-components/utils@1.0.0-beta.4) (2019-11-28)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.3](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.2...@medly-components/utils@1.0.0-beta.3) (2019-11-13)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.2](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.0.0-beta.1...@medly-components/utils@1.0.0-beta.2) (2019-11-13)

**Note:** Version bump only for package @medly-components/utils





# [1.0.0-beta.1](https://github.com/medlypharmacy/medly-components/compare/@medly-components/utils@1.2.0...@medly-components/utils@1.0.0-beta.1) (2019-11-13)


### Bug Fixes

* **all:** remove changelog for pre release setup ([0795f5c](https://github.com/medlypharmacy/medly-components/commit/0795f5c139b529c3f6b65442ff71a8826ecf2c56))
* fixed various issues in date picker ([00fed84](https://github.com/medlypharmacy/medly-components/commit/00fed842b6d363f817de2c8703785c62557cf50d))


### Features

* **all:** remove s3 upload command from all packages ([89ea085](https://github.com/medlypharmacy/medly-components/commit/89ea085082829cba2098342c9793a43e2bc2e24a))
* **core:** add capability to change option in Single select through arrow keys ([fd864b9](https://github.com/medlypharmacy/medly-components/commit/fd864b9eb103bc042086c51236f10222cfc57367))
* added capitalize function to capitalize first letter of the string ([2613604](https://github.com/medlypharmacy/medly-components/commit/261360434dcd02de4cde438350d00696323d16d5))
* added DatePicker and DateRangePicker component ([5dc6d38](https://github.com/medlypharmacy/medly-components/commit/5dc6d38f66f6d1d530f830da13a9624715f6f704))
