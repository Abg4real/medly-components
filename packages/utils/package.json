{
    "name": "@medly-components/utils",
    "version": "1.0.0-beta.9",
    "side-effects": false,
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "files": [
        "dist/"
    ],
    "publishConfig": {
        "access": "public"
    },
    "license": "MIT",
    "scripts": {
        "typecheck": "tsc -p tsconfig.json",
        "build:js": "cross-env BABEL_ENV=production babel --root-mode upward src --out-dir dist --extensions .ts,.tsx --no-comments",
        "build:types": "tsc -p tsconfig.build.json",
        "prebuild": "yarn typecheck",
        "build": "yarn build:js --delete-dir-on-start ",
        "postbuild": "yarn build:types",
        "watch": "run-p \"build:types -w\" \"build:js --watch\""
    },
    "peerDependencies": {
        "react": "^16.8.6",
        "styled-components": "^4.2.0"
    },
    "dependencies": {
        "@medly-components/theme": "^1.0.0-beta.8"
    }
}
