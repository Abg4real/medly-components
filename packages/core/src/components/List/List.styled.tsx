/* stylelint-disable  no-descending-specificity*/
import { defaultTheme, SpacingTheme } from '@medly-components/theme';
import { css, styled } from '@medly-components/utils';
import { Props } from './types';

const horizontal = (spacings: SpacingTheme) => css`
    li {
        margin: 0px ${spacings.XS};
        &:last-child {
            margin-right: ${spacings.S};
        }
        &:first-child {
            margin-left: ${spacings.S};
        }
    }
`;

const vertical = (spacings: SpacingTheme) => css`
    li {
        margin: ${spacings.XS} 0px;
        &:last-child {
            margin-bottom: ${spacings.S};
        }
        &:first-child {
            margin-top: ${spacings.S};
        }
    }
`;

export const ListStyled = styled('ul').attrs(({ theme: { spacing: spacings } }) => ({ spacings }))<Props>`
    padding: 0;
    margin: 0;
    display: flex;
    list-style: none;
    flex-direction: ${({ variant }) => (variant === 'horizontal' ? 'row' : 'column')};

    ${({ spacings, variant }) => variant === 'horizontal' && horizontal(spacings)};
    ${({ spacings, variant }) => variant === 'vertical' && vertical(spacings)};
`;

ListStyled.defaultProps = {
    theme: defaultTheme
};
