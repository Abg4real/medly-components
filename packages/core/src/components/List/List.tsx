import { WithStyle } from '@medly-components/utils';
import React, { SFC } from 'react';
import { ListStyled } from './List.styled';
import { Props } from './types';

export const List: SFC<Props> & WithStyle = React.memo(props => (
    <ListStyled {...props}>
        {React.Children.map(props.children, (c, i) => {
            return <li key={i}>{c}</li>;
        })}
    </ListStyled>
));

List.displayName = 'List';
List.Style = ListStyled;
List.defaultProps = {
    variant: 'vertical'
};
