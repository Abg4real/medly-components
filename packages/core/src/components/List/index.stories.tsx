import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import React from 'react';
import { wInfo } from '../../utils/wInfo';
import Text from '../Text';
import List from './List';
import { Props } from './types';

const variant: Array<Props['variant']> = ['horizontal', 'vertical'];

storiesOf('Components', module).add(
    'List',
    () => (
        <List variant={select('Variant', variant, 'vertical')}>
            <Text> Item 1</Text>
            <Text> Item 2</Text>
            <Text> Item 3</Text>
        </List>
    ),
    wInfo('List component to show show the list either horizontaly or vertically')
);
