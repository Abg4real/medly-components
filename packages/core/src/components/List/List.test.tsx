import { TestUtils } from '@medly-components/utils';
import React from 'react';

import { List } from './List';

describe('List component', () => {
    it('should render horizontal list', () => {
        const { container } = TestUtils.render(
            <List variant="horizontal">
                <p>item 1</p>
                <p>item 2</p>
            </List>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render vertical list', () => {
        const { container } = TestUtils.render(
            <List variant="vertical">
                <p>item 1</p>
                <p>item 2</p>
            </List>
        );
        expect(container).toMatchSnapshot();
    });
});
