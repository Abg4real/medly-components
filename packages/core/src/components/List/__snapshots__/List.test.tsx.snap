// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should render horizontal list 1`] = `
.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 li {
  margin: 0 0.4rem;
}

.c0 li:last-child {
  margin-right: 0.8rem;
}

.c0 li:first-child {
  margin-left: 0.8rem;
}

<div>
  <ul
    class="c0"
  >
    <li>
      <p>
        item 1
      </p>
    </li>
    <li>
      <p>
        item 2
      </p>
    </li>
  </ul>
</div>
`;

exports[`List component should render vertical list 1`] = `
.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 li {
  margin: 0.4rem 0;
}

.c0 li:last-child {
  margin-bottom: 0.8rem;
}

.c0 li:first-child {
  margin-top: 0.8rem;
}

<div>
  <ul
    class="c0"
  >
    <li>
      <p>
        item 1
      </p>
    </li>
    <li>
      <p>
        item 2
      </p>
    </li>
  </ul>
</div>
`;
